### ISSUE TITLE
[Tree-item] No focus present when navigating via keyboard

### ISSUE THREAD

**Issue by rpanichakit9541** [Esri/calcite-components#6376]
### Summary

When using keyboard navigation, the `tree` and `tree-item` components do not have a focus present.

### Actual Behavior

No focus present on the components.

### Expected Behavior

A focus should be present on the `tree` and `tree-item` components.

### Reproduction Sample

https://codepen.io/rpanichakit/pen/wvxEJvG

### Reproduction Steps

Use your keyboard to navigate through the calcite-tree-item, and notice that there's no focus present.

### Reproduction Version

1.0.3

### Regression?

beta.99

### Esri team

ArcGIS Online

**Comment by geospatialem** [comment]
@rpanichakit9541 Thanks for reaching out! Accessibility for tree structures follows the [tree role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tree_role), which would expand/collapse `tree-item`s with the `Arrow right`/`Arrow left` keys respectively. We've documented a similar keyboard navigation in the components [keyboard navigation](https://developers.arcgis.com/calcite-design-system/components/tree/#keyboard-navigation).

However, there does seem to be a different with the component's focus, which occurred somewhere between `beta.99` and `1.0.3`. @jcfranco This might be a good one to tackle either in the upcoming patch of February release - WDYT?


Example behavior with focus below:

### beta.99
![beta 99-tree](https://user-images.githubusercontent.com/5023024/215594340-738300a3-be3a-4b4a-87a7-8e467f7f016f.gif)

### 1.0.3
![1 0 3-tree](https://user-images.githubusercontent.com/5023024/215594689-e3f4eafa-64ec-4761-b2ce-bc1ecf367ec1.gif)

**Comment by rpanichakit9541** [comment]
@geospatialem - Thank you for the quick reply.  I apologize I missed that keyboard navigation section.  Feel free to reuse this for the focus issue or close it.

**Comment by geospatialem** [comment]
Updated the issue labels, as the focus functionality works as-expected in Firefox, but does not display in Chrome or Microsoft Edge. Identified when @Elijbet was working through a fix.

**Comment by geospatialem** [comment]
Verified in Storybook with Chrome, Edge, and Firefox.
